client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -492,14 +494,17 @@ private void reconnect(
             if (connectionFailureCause != null) {
                 log.debug("connection attempt failed", connectionFailureCause);
             }
-            final long reconnectInterval = Optional.ofNullable(clientOptions)
-                    .map(o -> o.getReconnectInterval())
-                    .orElse(Constants.DEFAULT_RECONNECT_INTERVAL_MILLIS);
-            log.trace("scheduling new connection attempt in {}ms ...", reconnectInterval);
-            // give Vert.x some time to clean up NetClient
-            vertx.setTimer(reconnectInterval, tid -> {
+            final long reconnectInterval = Math.min(clientConfigProperties.getReconnectMaxDelay(),
+                    clientConfigProperties.getReconnectMinDelay()
+                            + reconnectAttempt * clientConfigProperties.getReconnectDelayIncrement());
```

Comment:
it seems that you do not want to use the usual *exponential back-off* algorithm here. Is there a particular reason for that? I am not sure if incrementing the delay in steps of 100ms will actually help the peer to relief from its problems ...

Reply:
Right, I've changed the implementation, now applying an exponential backoff with jitter, the delay being a random between the minDelay and the value exponentially increasing with every attempt. The delay is capped by the maxDelay.